go.mod

```diff
@@ -46,11 +46,11 @@ require (
 	github.com/ugorji/go/codec v1.2.12 // indirect
 	golang.org/x/arch v0.13.0 // indirect
 	golang.org/x/net v0.34.0 // indirect
-	golang.org/x/sys v0.29.0 // indirect
-	golang.org/x/text v0.21.0 // indirect
+	golang.org/x/sys v0.30.0 // indirect
+	golang.org/x/text v0.22.0 // indirect
 	google.golang.org/protobuf v1.36.2 // indirect
 )
 
 go 1.21.0
 
-toolchain go1.23.4
+toolchain go1.24.0
```

Comment:
There is a coverage bug here (toolchain issue?). 

This line causes a lot of coverage to be missing even if the toolchain is already 1.24.0. But go.mod reference says it 'Only takes effect when the module is the main module and the default toolchain is older than the suggested toolchain.' so I understand it should be a no-op? 

I am investigating more on this.